src/engine/enginepregain.cpp

```diff
@@ -88,17 +88,17 @@ void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
         // full process for one second.
         // So we need to alter gain each time ::process is called.
 
-        const float fullReplayGainBoost =
-                fReplayGain * static_cast<float>(s_pReplayGainBoost->get());
+        const CSAMPLE_GAIN fullReplayGainBoost =
+                fReplayGain * static_cast<CSAMPLE_GAIN>(s_pReplayGainBoost->get());
 
         // This means that a ReplayGain value has been calculated after the
         // track has been loaded
-        constexpr float kFadeSeconds = 1.0;
+        constexpr float kFadeSeconds = 1.0f;
 
         if (m_bSmoothFade) {
             float seconds = static_cast<float>(m_timer.elapsed().toDoubleSeconds());
-            if (seconds < kFadeSeconds) {
-                // Fade smoothly
+            if (seconds < kFadeSeconds && m_dSpeed != 0.0 && m_dOldSpeed != 0.0) {
```

Comment:
does this mean that once the replaygain is calculated, mixxx will now change the gain while the track is playing back?

Reply:
These condition applies the replay gain instantly without fading if the track is or was stopped. If it plays, fading is done. 
The logic that a analyzer replay gain is not applied when playing is handled here: https://github.com/mixxxdj/mixxx/blob/59ca3a25b768c44ac0934ed4cb4504cac52bde54/src/mixer/basetrackplayer.cpp#L711